### ISSUE TITLE
SkyCoord applies units twice sometimes

### ISSUE THREAD

**Issue by jobovy** [jobovy/galpy#325]
In certain cases when running with ``astropy-units = True`` in the configuration file, ``Orbit.SkyCoord`` output applies the units twice, so for example distances become kpc^2 (reported by Jean-Francois Robitaille).

Simple example which does seem to work
```
from galpy.orbit import Orbit
import astropy.units as u
o= Orbit([1.,0.1,1.1,0.1,0.2,0.])
o.SkyCoord().distance.unit
# kpc
```

However, the following doesn't work
```
o = Orbit(vxvv=[0.,0.,0.,0.,0.,0.],radec=True)
o.SkyCoord().distance.unit
# kpc^2
```

Must be because the ``SkyCoord`` distance is obtained using ``o.dist()`` and then multiplied by ``unit.kpc`` and ``o.dist`` already returns a number with unit ``unit.kpc``. However, I'm not sure why the first one of these examples does work, because ``o.dist`` should *always* return something with a unit in this case...

**Comment by jobovy** [comment]
Difference appears to be that in the first case
```o._roSet
# False
```
so methods like ``o.dist`` return a distance in kpc, but for some reason don't apply the unit; *that* also seems like a bug.

**Comment by jobovy** [comment]
Closed in 6cc018ba85ca74fba5875a8475a1f5b4e922972c
